packages/quill/src/blots/block.ts

```diff
@@ -74,11 +92,12 @@ class Block extends BlockBlot {
   }
 
   insertBefore(blot: Blot, ref?: Blot | null) {
-    const { head } = this.children;
     super.insertBefore(blot, ref);
-    if (head instanceof Break) {
-      head.remove();
-    }
+    this.children.forEach((child) => {
+      if (child instanceof Break) {
+        child.optimize();
+      }
+    });
```

Comment:
I found it was necessary to add these calls to `optimize` here since it seems that optimize is not called when using the editor methods to modify text (i.e. `applyDelta`, `insertText`, etc.). This appears to be the case because [the existing call to `optimize` in `ScrollBlot`](https://github.com/slab/parchment/blob/main/src/blot/scroll.ts#L174) depends on there being 1 or more mutations causing the change, which only happens when the editor contents is changed through some user action in the browser.

If there is some better way of accomplishing this than what I've done here, please let me know.